### PR TITLE
feat: document open-api lib

### LINKED ISSUE #111: Document @ts-rest/open-api
This is potentially a _huge_ aspect of ts-rest, it would be awesome to document, possibly with some interactive examples how the lib works.

@ghoullier I'm happy to have you help out with this, mind that I'm not _particularly_ considering any changes to the @ts-rest/open-api lib to be breaking until we release it on the docs 🤔 

If you don't have time that's completely OK I can put it on the backlog!

### PR DESCRIPTION
**Pull request by oliverbutler**
Finally document this lib!!

closes #111

### CODE REVIEW

File: apps/docs/docs/open-api.mdx [hunk 1]
```diff
@@ -0,0 +1,140 @@
...
+
+No kidding, that's it 🥳 You can now serve this document however you want, use it for CodeGen for your non-TS clients, or use it to generate a Swagger UI!
+
+:::info
+Our OpenAPI generator uses `@anatine/zod-openapi` to geneate open-api docs for your Zod schemas, so make sure to use Zod everywhere on your API to get the best experience.
+:::
+
+## Serving a Swagger UI
+
+In Express use `swagger-ui-express`:
```
Comment (Gabrola) [inline apps/docs/docs/open-api.mdx:33]: We can use tabs (Express, NestJS) for a nicer UX like we have on the Quickstart page

File: apps/docs/docs/open-api.mdx [hunk 2]
```diff
@@ -0,0 +1,140 @@
...
+  //                   ^ Path for swagger
+
+  await app.listen(3000);
+}
+bootstrap();
+```
+
+Don't worry if you don't use express or Nest, whatever library you want to use is OK our OpenAPI returns a plain JSON object which is fully compliant with the OpenAPI spec.
+
+## Enabling `operationId`'s (Reccomended!)
```
Comment (Gabrola) [inline apps/docs/docs/open-api.mdx:79]: typo hehe

```suggestion
## Enabling `operationId`'s (Recommended!)
```

**Review by Gabrola — APPROVED** [review]
Finally! Thank you! ❤

File: apps/docs/docs/open-api.mdx [hunk 3]
```diff
@@ -0,0 +1,140 @@
...
+    title: 'Posts API',
+    version: '1.0.0',
+  },
+});
+```
+
+No kidding, that's it 🥳 You can now serve this document however you want, use it for CodeGen for your non-TS clients, or use it to generate a Swagger UI!
+
+:::info
+Our OpenAPI generator uses `@anatine/zod-openapi` to geneate open-api docs for your Zod schemas, so make sure to use Zod everywhere on your API to get the best experience.
```
Comment (michaelangrivera) [inline apps/docs/docs/open-api.mdx:28]:
```suggestion
Our OpenAPI generator uses `@anatine/zod-openapi` to generate open-api docs for your Zod schemas, so make sure to use Zod everywhere on your API to get the best experience.
```

File: apps/docs/docs/guides/non-ts-clients.mdx [hunk 4]
```diff
@@ -0,0 +1,19 @@
+# Without TS Clients
+
+In a brownfield code base it's common that you may not have the luxury of having Typescript end to end, possibly you've got a Android/iOS/Flutter app that's written in Java/Kotlin/Swift and you're looking to add Typescript to your backend.
```
Comment (michaelangrivera) [inline apps/docs/docs/guides/non-ts-clients.mdx:3]:
```suggestion
In a brownfield code base, it's common that you may not have the luxury of having Typescript end-to-end. For instance, you might have an Android/iOS/Flutter app written in Java/Kotlin/Swift and want to add Typescript to your backend.
```

**Review by arivera-xealth — APPROVED** [review]

**Review by michaelangrivera — APPROVED** [review]

**Pull request merged**